bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/TelescopePosTableWidget.java

```diff
@@ -594,32 +623,28 @@ private void startWatchingEnv() {
         _dataObject.addPropertyChangeListener(TargetObsComp.TARGET_ENV_PROP, envChangeListener);
     }
 
+    /**
+     * Notify the TargetSelection object when a row in the table has been selected.
+     * This can either be a target row, or a group row.
+     */
     private void notifySelect(final TableData.Row row) {
-        final SPTarget target = row.target().getOrNull();
-        if (target != null) {
+        // Only one of these cases will hold.
+        row.target().foreach(target -> {
             stopWatchingSelection();
-            try {
-                TargetSelection.set(_env, _obsComp, target);
-            } finally {
-                startWatchingSelection();
-            }
-        } else {
-            final GuideGroup group = row.group().map(Tuple2::_2).getOrNull();
-            if (group != null) TargetSelection.setGuideGroup(_env, _obsComp, group);
-        }
+            TargetSelection.setTargetForNode(_env, _obsComp, target);
+            startWatchingSelection();
```

Comment:
Maybe these things are better wrapped in the `try` block to avoid leaving the editor in a bad state if an exception is thrown.